source/retryable-writes/retryable-writes.rst

```diff
@@ -494,7 +506,7 @@ The above rules are implemented in the following pseudo-code:
        * is very rare, and likely means that the cluster is in the midst of a
        * downgrade. */
       if ( ! isRetryableWritesSupported(server)) {
-        throw originalError;
+        throw currentError;
```

Comment:
I beleive this should be previousError too.

Reply:
(adding another comment because I accidentally approved.)

Reply:
Originally added in [SPEC-972](https://jira.mongodb.org/browse/SPEC-972) , the comment for this block says 

> If the server selected for retrying is too old, throw the original error

What is the purpose for changing the logic here?

Reply:
Using currentError here is no longer correct when there are multiple retrie. Imagine:
- Attempt 1 fails with NoWritesPerformed
- Attempt 2 fails with ShutdownInProgress
- Attempt 3 fails with NoWritesPerformed
- Attempt 4 selects a server that doesn't support retryable writes

We have to raise the previous non-NoWritesPerformed error which would be the ShutdownInProgress from attempt 2, not the one from attempt 3.